crates/router/src/core/payment_methods/cards.rs

```diff
@@ -216,6 +225,216 @@ pub async fn get_or_insert_payment_method(
     }
 }
 
+#[instrument(skip_all)]
+pub async fn generate_client_secret_or_add_payment_method(
+    state: routes::AppState,
+    req: api::PaymentMethodCreate,
+    merchant_account: &domain::MerchantAccount,
+    key_store: &domain::MerchantKeyStore,
+) -> errors::RouterResponse<api::PaymentMethodResponse> {
+    let db = &*state.store;
+    let merchant_id = &merchant_account.merchant_id;
+    let customer_id = req.customer_id.clone().get_required_value("customer_id")?;
+
+    if req.payment_method_data.is_some() {
+        add_payment_method_data(state, req, merchant_account, key_store).await
+    } else if req.card.is_some() || req.bank_transfer.is_some() || req.wallet.is_some() {
+        add_payment_method(state, req, merchant_account, key_store).await
```

Comment:
I can create client_secret and pass payment method data in the already existing field `card` in which case it will go to `else if` block creating another payment methods entry in db?

Reply:
the `else if` block is meant for backwards compatibility, so the current version of API can also be supported while also enabling client based calls using `payment_method_data`